crates/module-system/module-implementations/sov-evm/src/authenticate.rs

```diff
@@ -45,14 +45,35 @@ fn recover_evm_signer(
 }
 
 /// Creates the transaction details for an EVM transaction.
-fn create_evm_tx_details<S: Spec>(tx: &TransactionSigned) -> TxDetails<S> {
+fn create_evm_tx_details<S: Spec>(
+    tx: &TransactionSigned,
+) -> Result<(TxDetails<S>, TxHash), AuthenticationError> {
+    let tx_hash = TxHash::new(**tx.hash());
```

Comment:
Why did we move it here?
TxHash is not part of TxDetails and this move makes the function name not fully descriptive and complicates return value while saving 0 lines of code.

Reply:
Renamed to `create_evm_tx_details_and_hash`